packages/core/src/components/toggle-button/toggle-button.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  Component,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Prop,
+  Watch,
+} from '@stencil/core';
+import { BaseButton, BaseButtonProps } from '../button/base-button';
+import { ButtonVariant } from '../button/button';
+import { a11yBoolean } from '../utils/a11y';
+
+/**
+ * @since 2.0.0
+ */
+@Component({
+  tag: 'ix-toggle-button',
+  shadow: true,
+  styleUrl: './toggle-button.scss',
+})
+export class ToggleButton {
```

Comment:
Because this would look more like:
```
   Omit<
      Button,
      'type' | 'submitButtonElement' | 'dispatchFormEvents' | 'hostElement'
    >
```

Wich does not seem very elegant IMO.